src/Microsoft.AspNetCore.Server.Testing/Deployers/ApplicationDeployer.cs

```diff
@@ -82,14 +82,31 @@ protected void DotnetPublish(string publishRoot = null)
 
         protected void CleanPublishedOutput()
         {
-            try
+            var deletePublishedFolderEnvironmentVariableName = "DELETE_PUBLISHED_FOLDER";
+            var deleteLocallyPublishedFolder = true;
+            var setting = Environment.GetEnvironmentVariable(deletePublishedFolderEnvironmentVariableName);
+            if (string.Equals(setting, "false", StringComparison.OrdinalIgnoreCase))
```

Comment:
I would take either `false` or `0` to disable the setting. See https://github.com/aspnet/Hosting/blob/41b4e6017e5d2c67e3a3f031558ded23abd28d58/src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs#L45-L49.